### ISSUE TITLE
Miner Mining things less profitable

### ISSUE THREAD

**Issue by bwstandard** [RainbowMiner/RainbowMiner#2346]
I just watched two rigs go to KAS while there were options that were more profitable. Like NEXA and or DNX. I'm not sure why it does that nor do I know how to do the ol' 90s "hit the side of the TV" trick to force it to what it should be mining. 

Any idea on this?

**Comment by whitindllc** [comment]
Yes i noticed it wants to mine a algo with a 5 plus hour time to find and less value per day BTC vs mining something with a 1.1 min time to find and higher BTC per day value

**Comment by bwstandard** [comment]
I will also pile on that I have turned off a miner, algo combo for a specific card, my 3060 and it still keeps coming up and causing crashes. Not sure if that's a bug or not.

**Comment by whitindllc** [comment]
> I will also pile on that I have turned off a miner, algo combo for a specific card, my 3060 and it still keeps coming up and causing crashes. Not sure if that's a bug or not.

I figured out how to fix that.... i had the same issue... i went into the config file and i found a line called delay and i changed it to 45 and it fixed that issue
how did you turn off the miner?? you have to exclude them exactly as they are named .. i have issues with bzminer and teamblackminer resetting the MSI settings and defaulting my fans to 0 speed
so i avoid them... in the config there is a section called excludeminer name... you have to name them what they show up in the rainbowminer powershell applet... for me they were bzminer and teamblack

in the rainbowminer\bin directory they show somewhat what the names are... they will say amd-%name% or nvidia-%name% but that doesnt always match the powershell names

**Comment by bwstandard** [comment]
I have a miner card combo that I need off because the miner works great otherwise. I just toggled it off in the miner section of the dashboard

**Comment by whitindllc** [comment]
> I have a miner card combo that I need off because the miner works great otherwise. I just toggled it off in the miner section of the dashboard

So what i found was it sorta thinks in a simple way... cpu1 is cpu1 and 2 is 2 and 3 is this 3.... what was happening for me is when the program would shut off (and these are examples)
ttminer using cards 0 2 3 4 6 
it would then open lolminer before ttminer shut down all the way.... cards 2, 4 and 6 were still tied up in TTminer
lol miner started looking to start the same cards... 0 started but instead of 2 it would sometimes start 1 or 5 or 7 because the program saw them as cards  3 4 and 6 now..... once i set the delay to a higher number than 1 it completely stopped doing that....

**Comment by RainbowMiner** [comment]
There are some settings in config.txt, that reduce frequent switching - it's a measure against wild price swings. Also, there is the accuracy of some pools that might have an influence (you can reduce that impact with the switching prevention parameters below). Most of the time, it's better to have some hysteresis in place. You can adjust this hysteresis by changing the following parameters:

MinimumMiningIntervals = Minimum mining intervals after miner start, before switching can happen [default=1]
PoolSwitchingHysteresis = switching prevention against pool-to-pool hopping, hysteresis offset in percent [default=3]
MinerSwitchingHysteresis = switching prevention against on pool miner-to-miner hopping, hysteresis offset in percent [default=3]
SwitchingPrevention = adjust switching prevention: the higher, the less switching of miners (0 to disable) [default=2]
EnableFastSwitching = remove accuracy checks [default=0]

In my experience setting the PoolSwitchingHysteresis and MinerSwitchingHysteresis to 2 already speeds up switching

**Comment by whitindllc** [comment]
> There are some settings in config.txt, that reduce frequent switching - it's a measure against wild price swings. Also, there is the accuracy of some pools that might have an influence (you can reduce that impact with the switching prevention parameters below). Most of the time, it's better to have some hysteresis in place. You can adjust this hysteresis by changing the following parameters:
> 
> MinimumMiningIntervals = Minimum mining intervals after miner start, before switching can happen [default=1] PoolSwitchingHysteresis = switching prevention against pool-to-pool hopping, hysteresis offset in percent [default=3] MinerSwitchingHysteresis = switching prevention against on pool miner-to-miner hopping, hysteresis offset in percent [default=3] SwitchingPrevention = adjust switching prevention: the higher, the less switching of miners (0 to disable) [default=2] EnableFastSwitching = remove accuracy checks [default=0]
> 
> In my experience setting the PoolSwitchingHysteresis and MinerSwitchingHysteresis to 2 already speeds up switching

this is what that chunk of my config looks like
    "MinimumMiningIntervals": 1,
  "DisableExtendInterval": "0",
  "SwitchingPrevention": 2.0,
  "PoolSwitchingHysteresis": 2.0,
  "MinerSwitchingHysteresis": 2.0,
  "MaxRejectedShareRatio": 0.3,
  "MaxAllowedLuck": 3.0,
  "MaxTimeSinceLastBlock": "12h",
  "EnableFastSwitching": "0",
  "ForceStablePrice": "0",,
